### DISCUSSION TITLE
nerdctl build failed when "RUN ln -snf /usr/share/zoneinfo/Asia/Shanghai /etc/localtime && echo Asia/Shanghai > /etc/timezone" in Dockerfile

### DISCUSSION THREAD

**Discussion by navygg** [containerd/nerdctl#1950] [Q&A] [answered]
I installed lima+nerdctl as alternatives of docker/docker desktop. But when I executed "lima nerdctl build", I got failed result:

 > [stage-1 2/6] RUN ln -snf /usr/share/zoneinfo/Asia/Shanghai /etc/localtime && echo Asia/Shanghai > /etc/timezone:
------
Dockerfile:17
--------------------
  15 |
  16 |     ENV TZ=Asia/Shanghai
  17 | >>> RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
  18 |
--------------------
error: failed to solve: failed to compute cache key: failed to mount /var/lib/containerd/tmpmounts/containerd-mount1304185167: operation not permitted: unknown
time="2023-01-30T09:53:56Z" level=fatal msg="unrecognized image format"
make: *** [upload] Error 1

**Comment by AkihiroSuda** [comment]
Please provide the minimal reproducer and the version of nerdctl, BuildKit, and the Linux distribution.

**Reply by navygg** [reply]
my laptop: Macbook Pro
I installed lima by "brew install lima", with built-in integration for nerdctl.

brew info lima
==> lima: stable 0.11.1 (bottled), HEAD
Linux virtual machines

lima uname -a
Linux lima-default 5.10.0-13-amd64 https://github.com/lima-vm/lima/issues/1 SMP Debian 5.10.106-1 (2022-03-17) x86_64 GNU/Linux

limactl list
NAME STATUS SSH ARCH CPUS MEMORY DISK
default Running 127.0.0.1:60022 x86_64 2 4GiB 50GiB

cat Dockerfile
FROM busybox

ENV TZ=Asia/Shanghai
RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone

lima nerdctl build -f Dockerfile -t busybox:test .
[+] Building 3.3s (5/5) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                     0.1s
 => => transferring dockerfile: 151B                                                                                                                     0.1s
 => [internal] load .dockerignore                                                                                                                        0.1s
 => => transferring context: 2B                                                                                                                          0.0s
 => [internal] load metadata for docker.io/library/busybox:latest                                                                                        3.0s
 => CACHED [1/2] FROM docker.io/library/busybox@sha256:7b3ccabffc97de872a30dfd234fd972a66d247c8cfc69b0550f276481852627c                                  0.1s
 => => resolve docker.io/library/busybox@sha256:7b3ccabffc97de872a30dfd234fd972a66d247c8cfc69b0550f276481852627c                                         0.1s
 => ERROR [2/2] RUN ln -snf /usr/share/zoneinfo/Asia/Shanghai /etc/localtime && echo Asia/Shanghai > /etc/timezone                                       0.1s
------
 > [2/2] RUN ln -snf /usr/share/zoneinfo/Asia/Shanghai /etc/localtime && echo Asia/Shanghai > /etc/timezone:
------
Dockerfile:4
--------------------
   2 |
   3 |     ENV TZ=Asia/Shanghai
   4 | >>> RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
   5 |
--------------------
error: failed to solve: process "/bin/sh -c ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone" did not complete successfully: operation not permitted
FATA[0003] unrecognized image format

**Comment by navygg** [comment, marked as answer]
I switched to Ubuntu, then succeed.

lima uname -a
Linux lima-default 5.15.0-57-generic https://github.com/lima-vm/lima/issues/63-Ubuntu SMP Thu Nov 24 13:43:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

So this is not nerdctl's fault, may be there is bug in lima's debian vm.
